riscv_ctg/generator.py

```diff
@@ -122,8 +155,16 @@ def __init__(self,fmt,opnode,opcode,randomization, xl, fl,base_isa_str):
         base_isa = base_isa_str
         self.fmt = fmt
         self.opcode = opcode
+
+        if (opnode['isa'] == 'IP'):
+            init_rvp_ops_vals(OPS, VALS)
         self.op_vars = OPS[fmt]
-        self.val_vars = VALS[fmt]
+
+        try:
```

Comment:
For the sake of uniformity in the way the variables are obtained, declaring the values in the dictionaries as a string would work. Using a try-except block to support 2 different methods isn't necessary.

Reply:
It would also be helpful if the dictionaries can be updated in this file itself and not be updated via a function. That would help localise the information.

Reply:
Done in 776e17d203d76102e223da300482ae655ee36691